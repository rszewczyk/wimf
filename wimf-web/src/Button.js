// @flow
import React from "react";
import { css, merge } from "glamor";

type ButtonProps = {
  onClick: Event => void,
  children: any,
  primary?: boolean
};

const buttonStyle = css({
  fontSize: ".875rem",
  fontWeight: 700,
  cursor: "pointer",
  display: "inline-block",
  lineHeight: "1.125rem",
  padding: ".5rem 1rem",
  margin: 0,
  border: "1px solid transparent",
  verticalAlign: "middle",
  borderRadius: "4px",
  ":hover": {
    boxShadow: "inset 0 0 0 10rem rgba(0,0,0,.075)"
  }
});

const buttonPrimaryStyle = merge(buttonStyle, {
  color: "#fff",
  backgroundColor: "#0074d9"
});

export default function Button(props: ButtonProps) {
  return (
    <button
      {...(props.primary ? buttonPrimaryStyle : buttonStyle)}
      onClick={props.onClick}
    >
      {props.children}
    </button>
  );
}
