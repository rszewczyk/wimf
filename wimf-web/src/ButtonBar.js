// @flow
import React from "react";
import { css } from "glamor";

type ButtonBarProps = {
  children: any
};

const barStyle = css({
  "> *": {
    marginLeft: "0.5rem",
    marginRight: "0.5rem"
  },
  padding: "0.5rem"
});

export default function ButtonBar(props: ButtonBarProps) {
  return (
    <div {...barStyle}>
      {props.children}
    </div>
  );
}
