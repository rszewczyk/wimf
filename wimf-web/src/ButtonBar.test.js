import React from "react";
import renderer from "react-test-renderer";
import ButtonBar from "./ButtonBar";
import Button from "./Button";

it("renders correctly", () => {
  const oneBtn = renderer
    .create(<ButtonBar><Button>Submit</Button></ButtonBar>)
    .toJSON();
  expect(oneBtn).toMatchSnapshot();

  const threeBtn = renderer
    .create(
      <ButtonBar><Button>Submit</Button><Button>Cancel</Button></ButtonBar>
    )
    .toJSON();
  expect(threeBtn).toMatchSnapshot();
});
