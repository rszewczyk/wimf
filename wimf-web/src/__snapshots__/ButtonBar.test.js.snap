// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  data-css-or04er=""
>
  <button
    data-css-7ag5l5=""
    onClick={undefined}
  >
    Submit
  </button>
</div>
`;

exports[`renders correctly 2`] = `
<div
  data-css-or04er=""
>
  <button
    data-css-7ag5l5=""
    onClick={undefined}
  >
    Submit
  </button>
  <button
    data-css-7ag5l5=""
    onClick={undefined}
  >
    Cancel
  </button>
</div>
`;
